cli/deno.json

```diff
@@ -0,0 +1,6 @@
+{
+  "tasks": {
+    "compile": "deno compile --allow-run=open --target aarch64-apple-darwin --output dist/devx-logs index.ts",
```

Comment:
Only compile for M series macs, because as a department, that's what we use for local development.

Reply:
The resulting binary is pretty big for what it does (>100MB). We _could_ use bash instead of Deno, however we'd lose type-safety etc.

Reply:
It's because the Deno runtime has to be shipped within the binary. Can't be helped unfortunately (:

Reply:
If we're compiling to a specific arch anyway, is it worth using a compiled language like Rust or Go instead? Would likely cut down on the runtime size significantly.

Reply:
> If we're compiling to a specific arch anyway, is it worth using a compiled language like Rust or Go instead? Would likely cut down on the runtime size significantly.

Chose TS on the assumption that it's the most familiar across the department (other than possibly Scala). If we move to JS, we could probably distribute this via NPM, rather than homebrew?

Reply:
I haven't worked with it much, but `deno install` could be a nice middle-ground, which wouldn't require much change to the code? https://docs.deno.com/runtime/manual/tools/script_installer

Reply:
@bryophyta my understanding is that `install` would create a script that invokes the CLI using `deno`. That means that people would have to have deno installed in order to run the CLI, since the runtime is not included.
I don't think we should require people to install additional software when running CLIs (which is why I'm personally averse to CLIs that need, for example, the JVM to run). I would rather have a 100MB standalone executable that's easy to distribute - I don't see it as different from a huge Electron app which we've pretty much accepted as the norm. 
As Ash says, there are languages there are better suited to writing CLIs, but Deno is a good compromise imho, since TypeScript is pretty much understood by everyone at the G.

Reply:
We distribute our dev tooling via [homebew](https://github.com/guardian/homebrew-devtools). I'd be keen for this to also be distributed the same way, to reduce cognitive load. Homebrew formulae supports the [idea of dependencies](https://docs.brew.sh/Formula-Cookbook#specifying-other-formulae-as-dependencies), so we could craft this such that the raw TS files form the Homebrew artifact, and we have a dependency on Deno. Theoretically this would provide the same benefit as distributing a binary - the reduction of "works on my machine" symptoms.

I'd suggest we ship as is, and revise our approach once we understand what to optimise for (disk usage, performance, maintainability, etc.). Our work laptops have quite a lot of storage (1TB), and I'd be surprised if ~100MB is noticeable.

WDYT?